pkg/assets/create/creater.go

```diff
@@ -144,7 +147,15 @@ func create(manifests map[string]*unstructured.Unstructured, client dynamic.Inte
 	// In case we fail to find a rest-mapping for the resource, force to fetch the updated discovery on next run.
 	reloadDiscovery := false
 
-	for _, path := range sortedManifestPaths {
+	for i, path := range sortedManifestPaths {
+		if i > 0 {
+			select {
+			case <-ctx.Done():
+				return ctx.Err(), false
```

Comment:
i'm not fan of cancelling in middle of the bulk run. i would rather give the loop the extra time to potentially get positive outcome (maybe it will succeed this run) instead of cancelling and error out.

Reply:
> i would rather give the loop the extra time to potentially get positive outcome (maybe it will succeed this run)...

Then don't cancel it?  I don't think we want to re-interpret "please give up and gently shut down" as "I'm starting to get bored, but go ahead and run for another several minutes if you think it might work" ;).

Reply:
> i'm not fan of cancelling in middle of the bulk run. i would rather give the loop the extra time to potentially get positive outcome (maybe it will succeed this run) instead of cancelling and error out.

I may let this play out.  I wouldn't gate it on that if above though.

Reply:
> I wouldn't gate it on that if above though.

So cancel immediately if requested by the `Context`, even if we've done nothing?